tests/Fixer/Comment/SingleLineCommentStyleFixerTest.php

```diff
@@ -0,0 +1,321 @@
+//',
+                '<?php
+/***
+ *
+ */',
+            ],
+            [
+                '<?php
+
+    // s',
+                '<?php
+
+    /***
+s    *
+     */',
+            ],
+
+            // Untouched cases
+            [
+                '<?php
+$a = 1; /* in code */ $b = 2;
+',
+            ],
+            [
+                '<?php
+    /*
+     * in code 2
+     */ $a = 1;
+',
+            ],
+            [
+                '<?php
+/***
+ *
+ */ $a = 1;',
+            ],
+            [
+                '<?php
+    /***
+s    *
+     */ $a = 1;',
+            ],
+            [
+                '<?php
+    /*
+     * first line
+     * second line
+     */',
+            ],
+            [
+                '<?php
+    /*
+     * first line
+     *
+     * second line
+     */',
+            ],
+            [
+                '<?php
+    /*first line
+second line*/',
+            ],
+            [
+                '<?php /** inline doc comment */',
+            ],
+            [
+                '<?php
+    /**
+     * Doc comment
+     */',
+            ],
+            [
+                '<?php # test',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideHashCases
+     */
+    public function testHashCases($expected, $input = null)
+    {
+        $this->fixer->configure(['comment_types' => ['hash']]);
+        $this->doTest($expected, $input);
+    }
+
+    public function provideHashCases()
+    {
+        return [
+            [
+                '<h1>This is an <?php //echo 123;?> example</h1>',
+                '<h1>This is an <?php #echo 123;?> example</h1>',
+            ],
+            [
+                '<?php
+                    // test
+                ',
+                '<?php
+                    # test
+                ',
+            ],
+            [
+                '<?php
+                    // test1
+                    //test2
+                    // test3
+                    // test 4
+                ',
+                '<?php
+                    # test1
+                    #test2
+                    # test3
+                    # test 4
+                ',
+            ],
+
+            // Untouched cases
+            [
+                '<?php
+                    //#test
+                ',
+            ],
+            [
+                '<?php
+                    /*
+                        #test
+                    */
+                ',
+            ],
+            [
+                '<?php // a',
+                '<?php # a',
+            ],
+            [
+                '<?php /* start-end */',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideAllCases
+     */
+    public function testAllCases($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
```

Comment:
> Not having failure records of a scenario is not a valid reason to not test that scenario IMO.

I disagree. A test should (pick one or more):

1. Explain an intent
1. Demonstrate an intent or a functionality
1. Explore bad/unhappy paths
1. Show separation of responsabilities
1. Prove a bug

Which all sum up to have a purpose.

The work done in v2 for configuration is astonishing in term of OOP.
Now we can trust configuration bonds are well checked and enforced because of the new structure.
If and when a bug will be found in a `configure`, tests should be addressed in the context they belong.

The test you suggest would have two flaws:

1. It wouldn't have a purpose
1. It would test the right thing in the wrong place

> Although I would say having more tests can't hurt :)

Actually more tests can hurt: imagine a day when you write legit code, that make the test you suggest fails. If you don't know why the test was written, how do you pick a thoughtful choice?